### PR TITLE
CLN: clarify TypeError for IndexSlice argument to pd.xs

### LINKED ISSUE #35301: BUG: xs not working with slice
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

#### Code Sample, a copy-pastable example

```python
data = """
C1  C2 V
A1  0  10
A1  1  20
A2  0  2
A2  1  3
B1  0  2
B2  1  3
"""

import pandas as pd
from io import StringIO
df = pd.read_csv(StringIO(data), sep=' +').set_index(['C1', 'C2'])

df.xs(pd.IndexSlice['A1', :])
```

#### Problem description

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/***/lib/python3.7/site-packages/pandas/core/generic.py", line 3535, in xs
    loc, new_index = self.index.get_loc_level(key, drop_level=drop_level)
  File "/home/***/lib/python3.7/site-packages/pandas/core/indexes/multi.py", line 2835, in get_loc_level
    raise TypeError(key)
TypeError: ('A1', slice(None, None, None))

```

also similar code produce the same problem (`df.xs(('A1', slice(None)))`). Strangely this works:
```python
df = pd.DataFrame({'a': [1, 2, 3, 1], 'b': ['a', 'b', 'c', 'd'], 'v': [2, 3, 4, 5]}).set_index(['a', 'b'])
df.xs(pd.IndexSlice[1, :])
```

#### Output of ``pd.show_versions()``

<details>
INSTALLED VERSIONS
------------------
commit           : None
python           : 3.7.7.final.0
python-bits      : 64
OS               : Linux
OS-release       : 5.7.7-100.fc31.x86_64
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : it_IT.UTF-8
LOCALE           : it_IT.UTF-8

pandas           : 1.0.5
numpy            : 1.19.0
pytz             : 2019.2
dateutil         : 2.7.5
pip              : 20.1.1
setuptools       : 41.6.0
Cython           : 0.29.15
pytest           : 4.0.0
hypothesis       : None
sphinx           : 3.1.1
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : 4.4.0
html5lib         : 1.0.1
pymysql          : None
psycopg2         : None
jinja2           : 2.10
IPython          : 7.16.1
pandas_datareader: 0.8.0
bs4              : 4.7.1
bottleneck       : 1.2.1
fastparquet      : None
gcsfs            : None
lxml.etree       : 4.4.0
matplotlib       : 3.2.2
numexpr          : 2.7.1
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pytables         : None
pytest           : 4.0.0
pyxlsb           : None
s3fs             : 0.4.2
scipy            : 1.5.1
sqlalchemy       : None
tables           : 3.5.2
tabulate         : 0.8.5
xarray           : 0.12.1
xlrd             : 1.2.0
xlwt             : 1.1.2
xlsxwriter       : None
numba            : 0.48.0

</details>

### PR DESCRIPTION
**Pull request by arw2019**
- [x] closes #35301 
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

As per discussion in #35301 we do not support `IndexSlice` arguments to `xs`. This PR aims to clarify the `TypeError` thrown when that happens.

I created a separate issue to house the discussion re: functionality of `loc` vs `xs` (#35418).

### CODE REVIEW

File: doc/source/whatsnew/v1.2.0.rst [hunk 1]
```diff
@@ -114,7 +114,7 @@ Missing
 MultiIndex
 ^^^^^^^^^^
 
--
+- Bug in `DataFrame.xs` when used with `IndexSlice` raises ``TypeError`` with message `Expected label or tuple of labels` (:issue:`35301`)
```
Comment (jreback) [inline doc/source/whatsnew/v1.2.0.rst:117]: can you use ``` :meth:`DataFrame.xs` ``` and same for ```IndexSlice```

File: pandas/core/generic.py [hunk 2]
```diff
@@ -3477,7 +3477,10 @@ class   animal   locomotion
 
         index = self.index
         if isinstance(index, MultiIndex):
-            loc, new_index = self.index.get_loc_level(key, drop_level=drop_level)
+            try:
+                loc, new_index = self.index.get_loc_level(key, drop_level=drop_level)
+            except TypeError:
+                raise TypeError(f"Expected label or tuple of labels, got {key}")
```
Comment (jreback) [inline pandas/core/generic.py:3483]: can you add `from e` (need to add the the except as well)

File: pandas/tests/test_multilevel.py [hunk 3]
```diff
@@ -1258,6 +1258,16 @@ def test_level_with_tuples(self):
         tm.assert_frame_equal(result, expected)
         tm.assert_frame_equal(result2, expected)
 
+        # GH 35301
```
Comment (jreback) [inline pandas/tests/test_multilevel.py:1261]: can you make this a new test

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/test_multilevel.py [hunk 4]
```diff
@@ -1280,6 +1280,26 @@ def test_level_with_tuples(self):
         tm.assert_frame_equal(result, expected)
         tm.assert_frame_equal(result2, expected)
 
+    def test_xs_with_IndexSlice(self):
+        # GH 35301
```
Comment (jreback) [inline pandas/tests/test_multilevel.py:1284]: can you move to: pandas/tests/indexing/multiindex/test_xs.py

**Review by jreback — CHANGES_REQUESTED** [review]
if you can move the test; merge master and ping on green.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @arw2019 very nice (failure is unrelated)

**Comment by arw2019** [comment]
great! thanks @jreback for reviewing

**Pull request merged**